### PR TITLE
Add help method for SGEN.

### LINKED ISSUE #23375: Add help method in SGEN
The help method just shows "in development". Need fill the right content.

### PR DESCRIPTION
**Pull request by huanwu**
Fix #23375

@shmao @zhenlan @mconnew 

The generated help information is like:
```
Generates serialization code for use with XmlSerializer.
The utility allows developers to pre-generate code for serialization
building and deploying the assemblies with the application.


Usage: dotnet dotnet-Microsoft.XmlSerializer.Generator [[/assembly:<assembly name>] | [<assembly file location>]]
      [/type:] [/debug].

  Developer options:
     /assembly:   Assembly location or display name. Short form is '/a:'.
     /type:       Generate code for serialization/deserialization of the
                  specified type from the input assembly. Short form is '/t:'.
     /proxytypes  Generate serialization code only for proxy classes and web
                  method parameters. Short form is '/p'.
     /debug       Generate image which can be used under a debugger.
                  Short form is '/d'.
     /force       Forces overwrite of a previously generated assembly.
                  Short form is '/f'.
     /out:        Output directory name (default: target assembly location).
                  Short form is '/o:'.
  Miscellaneous options:
     /? or /help  Show this message
```

### CODE REVIEW

File: src/System.Private.Xml/src/Resources/Strings.resx [hunk 1]
```diff
@@ -3402,4 +3402,51 @@
   <data name="FailLoadAssemblyUnderPregenMode" xml:space="preserve">
     <value>"Fail to load assembly {0} or {0} doesn't exist under PreGen Mode.</value>
   </data>
+  <data name="HelpDescription" xml:space="preserve">
+    <value>Generates serialization code for use with XmlSerializer.
```
Comment (zhenlan) [inline src/System.Private.Xml/src/Resources/Strings.resx]: The tool takes in an assembly and output an assembly. Code generation is just implementation detail, which we should not mention as part of description.

File: src/Microsoft.XmlSerializer.Generator/src/Sgen.cs [hunk 2]
```diff
@@ -346,8 +346,18 @@ private void WriteHeader()
 
         private void WriteHelp()
         {
-            //TBD
-            Console.WriteLine("In Development");
+            Console.Out.WriteLine(SR.Format(SR.HelpDescription));
+            Console.Out.WriteLine(SR.Format(SR.HelpUsage, this.GetType().Assembly.GetName().Name));
+            Console.Out.WriteLine(SR.Format(SR.HelpDevOptions));
+            Console.Out.WriteLine(SR.Format(SR.HelpAssembly, "/assembly:", "/a:"));
+            Console.Out.WriteLine(SR.Format(SR.HelpType, "/type:", "/t:"));
```
Comment (zhenlan) [inline src/Microsoft.XmlSerializer.Generator/src/Sgen.cs]: Are all these options supported by upcoming release? If not, we should not include options that we don't support yet.

File: src/Microsoft.XmlSerializer.Generator/src/Sgen.cs [hunk 3]
```diff
@@ -346,8 +346,18 @@ private void WriteHeader()
 
         private void WriteHelp()
         {
-            //TBD
-            Console.WriteLine("In Development");
+            Console.Out.WriteLine(SR.Format(SR.HelpDescription));
+            Console.Out.WriteLine(SR.Format(SR.HelpUsage, this.GetType().Assembly.GetName().Name));
```
Comment (zhenlan) [inline src/Microsoft.XmlSerializer.Generator/src/Sgen.cs]: Is the "dotnet-" in  "dotnet-Microsoft.XmlSerializer.Generator" required?

File: src/Microsoft.XmlSerializer.Generator/src/Sgen.cs [hunk 4]
```diff
@@ -346,8 +346,18 @@ private void WriteHeader()
 
         private void WriteHelp()
         {
-            //TBD
-            Console.WriteLine("In Development");
```
Comment (zhenlan) [inline src/Microsoft.XmlSerializer.Generator/src/Sgen.cs]: We should have name and version as part of output header. For example,

```
Microsoft .NET Core Xml Serialization Generation Utility
Version: 1.0.0-preview1-xxxx
```

I am wondering, what's the message if someone runs this tool while .NET Core 2.0 is not installed on the machine.

File: src/Microsoft.XmlSerializer.Generator/src/Sgen.cs (hunk 2 above)
Comment (huanwu) [reply src/Microsoft.XmlSerializer.Generator/src/Sgen.cs]: Will remove debug. All others are supported.

**Pull request merged**